### ISSUE TITLE
.net core 5 single file error

### ISSUE THREAD

**Issue by gwhzh21** [dotnet/orleans#6898]
client = new ClientBuilder()
                        .UseLocalhostClustering(gatewayPort:gatewayPort)
                        .Build();

Unhandled exception. System.NotSupportedException: CodeBase is not supported on assemblies loaded from a single-file bundle.
   at System.Reflection.RuntimeAssembly.get_CodeBase()
   at Microsoft.Extensions.DependencyModel.DependencyContextLoader.GetNormalizedCodeBasePath(Assembly assembly)
   at Microsoft.Extensions.DependencyModel.DependencyContextLoader.GetDepsJsonPath(Assembly assembly)
   at Microsoft.Extensions.DependencyModel.DependencyContextLoader.LoadAssemblyContext(Assembly assembly, IDependencyContextReader reader)
   at Microsoft.Extensions.DependencyModel.DependencyContextLoader.Load(Assembly assembly)
   at Microsoft.Extensions.DependencyModel.DependencyContext.LoadDefault()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Extensions.DependencyModel.DependencyContext.get_Default()
   at Orleans.ApplicationPartManagerExtensions.AddFromDependencyContext(IApplicationPartManager manager, Assembly entryAssembly)
   at Orleans.ApplicationPartManagerExtensions.AddFromDependencyContext(IApplicationPartManager manager)
   at Orleans.ApplicationPartManagerExtensions.ConfigureDefaults(IApplicationPartManager applicationPartsManager)
   at Orleans.ClientBuilder.<>c.<Build>b__11_0(IApplicationPartManager parts)
   at Orleans.ClientBuilderExtensions.ConfigureApplicationParts(IClientBuilder builder, Action`1 configure)
   at Orleans.ClientBuilder.Build()

**Comment by ReubenBond** [comment]
This indeed looks like an issue with Orleans which we should fix for 3.x, thanks for reporting. As a workaround, you can try calling `builder.ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(TypeInYourAsm).Assembly).WithReferences())` on the client and silo.

**Comment by ReubenBond** [comment]
Did the above workaround work for you, @gwhzh21?
If not, please try adding the following PackageReference to any project which has this issue:

``` xml
<PackageReference Include="Microsoft.Extensions.DependencyModel" Version="5.0.0" />
```

**Comment by ReubenBond** [comment]
Fixed by #7197
